### PR TITLE
mkDoc

### PR DESCRIPTION
**Pull request by morganthomas**

### CODE REVIEW

**Review by MatthewCroughan — APPROVED** [review]
This is perfect.

File: internal/mkdoc.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  let deps = [ (texlive.combine { inherit (texlive) scheme-basic amsmath graphics hyperref; }) ];
+  in
+    stdenv.mkDerivation {
+      name = name;
+      src = src;
+      buildInputs = deps;
+      buildPhase = ''
+        mkdir -p $out
+        HOME=./. pdflatex ${inFile}.tex
+        HOME=./. pdflatex ${inFile}.tex
```
Comment (srid) [inline internal/mkdoc.nix:20]: Why run this twice?

Comment (morganthomas) [reply internal/mkdoc.nix:20]: TeX is a single pass compiler so it requires multiple runs to resolve forward references to labels and citations.

Comment (nixinator) [reply internal/mkdoc.nix:20]: I didn't know this, latex is like C! :-)

**Pull request merged**